lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=ManageIQ/manageiq#12345
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic
+    #
+    #
+    # === In ManageIQ/manageiq PR #1234
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq-api,manageiq-ui-classic \
+    #     including Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature,ManageIQ/manageiq#1234
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-api
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic
+    #
+    # TODO:  Handle the "self" case, where `manageiq` is also a TEST_REPO
+    #
+    # (maybe include a "self" helper as well?)
+    #
+    # === In ManageIQ/manageiq-ui-classic PR #1234:
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq#5678
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=manageiq#5678
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=manageiq-ui-classic#1234
+    #
+    class RunTest < Base
+      # The user calling the command
+      attr_reader :issuer
+
+      # The org of the issue this command originated from
+      attr_reader :org
+
+      # The (extra) repo(s) being targeted to be included in the test run
+      attr_reader :repos
+
+      # The repo(s) that will have the test suite run
+      attr_reader :test_repos
+
+      # The *-cross_repo-tests rugged instance
+      attr_reader :rugged_repo
+
+      # The arguments for the `run-test` command being called
+      attr_reader :value
+
+      restrict_to :organization
+
+      def self.test_repo_url
+        Settings.github.run_tests_repo.url
+      end
+
+      def self.test_repo_name
+        Settings.github.run_tests_repo.name
+      end
+
+      def self.test_repo_clone_dir
+        @test_repo_clone_dir ||= begin
+                                   url_parts = test_repo_url.split("/")[-2, 2]
+                                   repo_org  = url_parts.first
+                                   repo_dir  = test_repo_name
+                                   File.join(::Repo::BASE_PATH, repo_org, repo_dir)
+                                 end
+      end
+
+      # Name of the bot
+      def self.bot_name
+        Settings.github_credentials.username
+      end
+
+      # The new branch name for this particular run of the command (uniq)
+      def branch_name
+        @branch_name ||= "#{self.class.bot_name}-run-tests-#{issue.number}-#{SecureRandom.uuid}"
```

Comment:
I wonder if this should include the repo name as well or just remove the issue number?  issue number is just a number... e.g. in `miq-bot-1-01234abcdef` the `1` doesn't give you enough context, and doesn't help with uniqueness (since the uuid does that), so it's not contributing anything helpful to the naming by itself.

Reply:
Again, this was me mostly tweaking what @d-m-u had in her PR.

That said, if we move the `SecureRandom.uuid` to the beginning of the name, does that help?  Also, I probably should have added it as `#{issue.repo_name}_#{issue.number}`...

Reply:
oh the uuid bit is fine and I don't really have an opinion on the location within the string...I'm more question the issue number.  I think `#{issue.repo_name}_#{issue.number}` would work fine

Reply:
> oh the uuid bit is fine and I don't really have an opinion on the location within the string...

I was more suggesting that in it's current form, the `issue.number` and `uuid` get a bit muddled, and splitting them up to separate ends makes it a little more clear.

Reply:
ah ok....yeah I didn't see that, but now that you mention it I agree.